LeanOA/ForMathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Abs.lean

```diff
@@ -0,0 +1,236 @@
+    (abs a) ^ x = (star a * a) ^ (x / 2) := by
+  simp only [← abs_nnrpow_two_mul, mul_div_left_comm, ne_eq, OfNat.ofNat_ne_zero,
+    not_false_eq_true, div_self, mul_one]
+
+lemma sqrt_eq_real_sqrt (a : A) (ha : 0 ≤ a := by cfc_tac) :
+    CFC.sqrt a = cfcₙ Real.sqrt a := by
+  rw [sqrt_eq_iff _ (hb := cfcₙ_nonneg (A := A) (fun x _ ↦ Real.sqrt_nonneg x)),
+    ← cfcₙ_mul ..]
+  conv_rhs => rw [← cfcₙ_id (R := ℝ) a]
+  refine cfcₙ_congr fun x hx ↦ ?_
+  refine Real.mul_self_sqrt ?_
+  exact quasispectrum_nonneg_of_nonneg a ha x hx
+
+lemma abs_of_nonneg (a : A) (ha : 0 ≤ a := by cfc_tac) : abs a = a := by
+  rw [abs, ha.star_eq, sqrt_mul_self a ha]
+
+lemma abs_of_nonpos (a : A) (ha : a ≤ 0 := by cfc_tac) : abs a = -a := by
+  simp [← abs_neg a, abs_of_nonneg _ <| neg_nonneg.mpr ha]
+
+lemma abs_eq_norm (a : A) (ha : IsSelfAdjoint a := by cfc_tac) :
+    abs a = cfcₙ (‖·‖) a := by
+   simp only [abs, Real.norm_eq_abs, ← Real.sqrt_sq_eq_abs, sq]
+   have H : cfcₙ Real.sqrt (a * a) = cfcₙ (fun x ↦ √(x * x)) a := by
+       rw [← Function.comp_def, cfcₙ_comp a (f := fun x ↦ x * x) (g := fun x ↦ √x), cfcₙ_mul .., cfcₙ_id' ..]
+   rw [sqrt_eq_real_sqrt (ha := star_mul_self_nonneg a), ha.star_eq, H]
+
+protected lemma posPart_add_negPart (a : A) (ha : IsSelfAdjoint a := by cfc_tac) : abs a = a⁺ + a⁻ := by
+  rw [CFC.posPart_def, CFC.negPart_def, ← cfcₙ_add .., abs_eq_norm a ha]
+  exact cfcₙ_congr fun x hx ↦ (posPart_add_negPart x).symm
+
+lemma abs_sub_self (a : A) (ha : IsSelfAdjoint a := by cfc_tac) : abs a - a = 2 • a⁻ := by
+  nth_rw 2 [← CFC.posPart_sub_negPart a]
+  rw [CFC.posPart_add_negPart a]
+  abel
+
+lemma abs_add_self (a : A) (ha : IsSelfAdjoint a := by cfc_tac) : abs a + a = 2 • a⁺ := by
+  nth_rw 2 [← CFC.posPart_sub_negPart a]
+  rw [CFC.posPart_add_negPart a]
+  abel
+
+@[simp]
+lemma abs_abs (a : A) : abs (abs a) = abs a :=
+  abs_of_nonneg ..
+
+variable [StarModule ℝ A]
+
+lemma abs_smul_real (r : ℝ) (a : A) : abs (r • a) = |r| • abs a := by
+  have : 0 ≤ |r| • abs a := smul_nonneg (by positivity) abs_nonneg
+  rw [abs, CFC.sqrt_eq_iff _ _ (star_mul_self_nonneg _) this]
+  simp only [mul_smul_comm, smul_mul_assoc, abs_mul_self, star_smul]
+  match_scalars
+  simp
+
+lemma abs_smul_nnreal (r : ℝ≥0) (a : A) : abs (r • a) = r • abs a := by
+  simpa [NNReal.abs_eq] using abs_smul_real r a
+
+end Real
+
+section Complex
+
+variable [NonUnitalRing A] [TopologicalSpace A] [Module ℂ A]
+
+variable [StarRing A] [PartialOrder A] [StarOrderedRing A]
+variable [IsScalarTower ℂ A A] [SMulCommClass ℂ A A]
+variable [NonUnitalContinuousFunctionalCalculus ℂ (IsStarNormal : A → Prop)]
+
+open ComplexOrder in
+lemma cfcₙ_norm_sq_nonneg {f : ℂ → ℂ} {a : A} : 0 ≤ cfcₙ (fun z ↦ star (f z) * (f z)) a :=
+  cfcₙ_nonneg fun _ _ ↦ star_mul_self_nonneg _
+
+open ComplexOrder in
+lemma cfcₙ_norm_nonneg {a : A} : 0 ≤ cfcₙ (fun z : ℂ ↦ (‖z‖ : ℂ)) a :=
+  cfcₙ_nonneg fun _ _ ↦ by simp only [Complex.zero_le_real, norm_nonneg]
```

Comment:
Here's a trivial generalization, but I think it should actually be generalized further to `RCLike 𝕜` instead of `ℂ`. Moreover, this, and it's unital counterpart, should be placed in a more general file (probably we can have a `Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.RCLike` file).
```suggestion
lemma cfcₙ_norm_nonneg (f : ℂ → ℂ) (a : A) : 0 ≤ cfcₙ (fun z : ℂ ↦ (‖f z‖ : ℂ)) a :=
  cfcₙ_nonneg fun _ _ ↦ by simp [Complex.zero_le_real]
```
Again, for now, you can just put it in a `ForMathlib/Miscellany.lean` file in this repo.

I wonder how many of the results in this section can be generalized to `RCLike`. That would be nice if possible.